pkg/daemon/ceph/osd/device.go

```diff
@@ -438,16 +439,46 @@ func getStoreSettings(cfg *osdConfig) (map[string]string, error) {
 	return settings, nil
 }
 
-func writeConfigFile(cfg *osdConfig, context *clusterd.Context, cluster *mon.ClusterInfo, location string, prepareOnly bool) error {
+func WriteConfigFile(context *clusterd.Context, cluster *mon.ClusterInfo, kv *k8sutil.ConfigMapKVStore, osdID int, storeConfig config.StoreConfig, nodeName, location string) error {
+	scheme, err := config.LoadScheme(kv, config.GetConfigStoreName(nodeName))
+	if err != nil {
+		return fmt.Errorf("failed to load partition scheme: %+v", err)
+	}
+
+	config := &osdConfig{id: osdID, configRoot: context.ConfigDir, rootPath: getOSDRootDir(context.ConfigDir, osdID),
```

Comment:
so we are assuming it's a directory here until iterating through the partition scheme in the next step proves otherwise?  should we be using the `LoadOSDDirMap` function to verify its a directory and we already know about it?